### PR TITLE
Fix strict mode errors in built-ins/String

### PR DESCRIPTION
**Pull request by anba**
- Add missing "var" declarations and noStrict flags
- Remove with-statements (minor change in behaviour - primitive vs wrapper lookup, but does not seem too important for these tests)

Part of issue #35.

### CODE REVIEW

File: test/built-ins/String/prototype/indexOf/S15.5.4.7_A4_T1.js [hunk 1]
```diff
@@ -12,21 +12,17 @@ includes: [$FAIL.js]
 
 var __obj = {toString:function(){return "\u0041B";}}
 var __obj2 = {valueOf:function(){throw "intointeger";}}
-var __str = {str__:"ABB\u0041BABAB"};
+var __str = "ABB\u0041BABAB";
 
 //////////////////////////////////////////////////////////////////////////////
 //CHECK#1
-with(__str){
-    with(str__){
```
Comment (arv) [inline test/built-ins/String/prototype/indexOf/S15.5.4.7_A4_T1.js]: What were they thinking?

**Comment by arv** [comment]
LGTM

File: test/built-ins/String/prototype/indexOf/S15.5.4.7_A4_T1.js (hunk 1 above)
Comment (bterlson) [inline test/built-ins/String/prototype/indexOf/S15.5.4.7_A4_T1.js]: I feel like all those with statements must have been for something but I can't imagine what....

Comment (arv) [inline test/built-ins/String/prototype/indexOf/S15.5.4.7_A4_T1.js]: Maybe their codegen used them because it was simpler?

Anyway. Good riddance!

Comment (anba) [inline test/built-ins/String/prototype/indexOf/S15.5.4.7_A4_T1.js]: It seems to be some case of: Let's use additional language features to increase the test surface. :confused:

**Pull request merged**